### ISSUE TITLE
Cinnamon 5.4.2: "Window borders" option from Themes is gone; Window widgets are now weird

### ISSUE THREAD

**Issue by nick-s-b** [linuxmint/cinnamon#10867]
```
 * Cinnamon version (cinnamon --version)
Cinnamon 5.4.2
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Arch Linux  5.18.5-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 16 Jun 2022 20:40:45 +0000 x86_64 GNU/Linux
 * Graphics hardware *and* driver used
Nvidia 515.48.07 
 * 32 or 64 bit
64bit
 ```

**Issue**
I've been using the same Arc-like theme for many years but after 5.4 upgrade, I've noticed that Window widgets are now weird and take up more space. I've tried to adjust them and see if some other theme can fix them but that's no longer possible to adjust because the "Window borders" setting is gone from "Themes" pane in System settings.

Now:
![Screenshot at 2022-06-21 19-47-15](https://user-images.githubusercontent.com/7957590/174914727-fb164e83-dca8-47cb-bf42-9f11ac8bc162.png)

Before:
![Screenshot at 2022-06-21 19-47-34](https://user-images.githubusercontent.com/7957590/174914745-69d60837-bc72-4a87-8793-421e9a47c9d4.png)

**Steps to reproduce**
Install Cinnamon 5.4.x.

**Expected behaviour**
You should be able to change window border themes like before.

**Comment by mtwebster** [comment]
That may be the theme maintainer's attempt at making the titlebar for traditional windows to match those of client-decorated ones (we've considered this but decided against it for now because of the wasted space).

It definitely doesn't have to be this way:
![image](https://user-images.githubusercontent.com/262776/175040061-4d49ee86-e71a-4eae-8d82-e7da24944d5e.png)
This is mint-y - the window on top is cinnamon 5.4, underneath is a virtual machine with 5.2.7

The window borders selector is gone because borders and titlebar are now defined fully in the gtk theme, not a separate 'metacity' theme. They had their own proprietary language and were difficult to create and maintain - small adjustments or tweaks could be a huge pain. I think this is probably a welcomed change by most theme artists, as there is just a single theme to develop, which uses mostly normal css.

I would try and hit up the theme maintainer about the particular issue in the images you attached.

**Comment by jelabarre59** [comment]
You make it sound like the only way to get non-fugly/non-flat window borders is to install an entire theme, which may or may not have the icons and controls, etc that you want.   I have still been copying in a handful of Metacity themes' metacity-theme-3.xml files just to get window borders that have some actual texture and color to them (especially colors that differentiate between active and non-active windows).  As examples, I usually prefer Esco or Crux.

I also find I have to force overrides to GTK2 & GTK3 in order to have arrows on ALL my scrollbars.  So what I'm wondering is how to ensure these capabilities will continue to be available in the future when someone manages to break the old Window Border support.  Not an accusation or complaint, but a serious question on how to mix & match full-scale theme elements so our desktops don't look like someone bought into the whole Metro/TIFKAM/Material Design scam.

**Comment by Secret-chest** [comment]
This is better for _consistency_ as GTK+ headerbars will look the same as regular ones.

**Comment by jelabarre59** [comment]
Ah, you mean the ones where the former titlebar area is the exact same texture/color as the window's background?   Yeah, _really_ don't like those.

Which would make me ask then; is that a defined, separate part of the theme?   Are those colorless and featureless components addressable, and could they be tweaked or overridden within the theme definition?   I've tried doing searches for it, but as usual I'm probably using the wrong keywords to search on.

**Comment by mtwebster** [comment]
Gtk headerbars are referring to some windows that have buttons/controls on the very top edge, even along side the window controls. These have been themed in Gtk from their very inception (at least 7 or 8 years now I think), but in mint only a few of our newer tools use them (warpinator, hypnotix, etc...). The traditional windows are what have changed here - instead of having a separate theme just for window frame and titlebar, it's also part of the Gtk theme now.

The elements are still defined separately in the themes, so can be made to appear distinct from the normal content if desired. In the themes, look for identifiers like 'ssd' (server-side-decorated - formerly metacity themed), 'csd' (client-side-decorated - headerbar style), 'headerbar' and 'titlebar'.

I think it's much easier to modify this stuff if you try to choose a theme to modify that uses ['sas'](https://sass-lang.com/).

For example, Mint-Y: https://github.com/linuxmint/mint-themes/blob/master/src/Mint-Y/gtk-3.0/sass/_common.scss#L3078-L3223

All of the titlebar definitions are kept together. Once this gets 'compiled' it all becomes an impossible mess (at least to me, though I barely know enough css to break things).

**Comment by jelabarre59** [comment]
Interesting that the Fedora build of Cinnamon Desktop (5.2.7 in Fedora 36) doesn't have sass at all (no sass directories or .scss files in the global theme directories).  I haven't tested F37 as yet, so it may be in later Cinnamon builds.  (Yes, I know this repo is more oriented towards Linux Mint, but it seems most of the work on Cinnamon goes on here as well).

Probably better to use the Sass extension in VSCodium to work with the theme files then.  But being CSS-based I'm thinking user overrides may actually work.

**Comment by mtwebster** [comment]
Cinnamon itself doesn't use sass directly - it's just a tool people use to simplify theme creation. Before cinnamon can make use of the theme, it must be compiled into normal css. So themes, even if they use sass for development, won't actually ship them in the final package.

User overrides definitely work (~/.config/gtk-3.0)

**Comment by jelabarre59** [comment]
yep, VSCodium has an extension to compile Sass to plain CSS.  How _good_ it is I'd have to find out.
